qemu/tests/single_driver_update.py

```diff
@@ -0,0 +1,170 @@
+import re
+import logging
+from autotest.client.shared import error
+from virttest import utils_misc
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    This Test is mainly used as subtests
+    1) Boot up VM
+    2) Find the driver from the repo
+    3) record the driver version
+    4) Copy it to C:\ (If there are two copy the newest one)
+    5) Uninstall the driver/downgrade the driver
+    6) Reinstall the driver copied to C:\\tmp
+    7) Check the driver version whether same as original
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    def get_installation_logs(session):
+        tmp_debug = "%systemroot%\\DPINST.log,\
+                     %systemroot%\\inf\\setupapi.dev.log"
+        driver_logs = params.get("driver_debug_file", tmp_debug).split(',')
+        error.context("Get Driver installation log", logging.info)
+        for _file in driver_logs:
+            s, o = session.cmd_status_output("dir %s" % _file)
+            if not s:
+                output = session.cmd("type %s" % _file)
+                _file_host = _file.split("\\")[-1].strip()
+                _debug_log = open("%s/%s" % (test.resultsdir, _file_host), "w")
+                _debug_log.write("%s\n" % output)
+                _debug_log.close()
+
+    def backup_latest_folder(session, folders):
+        driver_repo_path = params.get("driver_repo_path")
+        tmp_folder = params.get("tmp_folder", 'C:\tmp')
```

Comment:
Suggest to add step description
error.context(Backup latest folder to %s % tmp_folder)